crypto/verificationhelper/sas.go

```diff
@@ -666,6 +666,14 @@ func (vh *VerificationHelper) onVerificationMAC(ctx context.Context, txn Verific
 				vh.cancelVerificationTxn(ctx, txn, event.VerificationCancelCodeUser, "failed to update device trust state after verifying: %w", err)
 				return
 			}
+
+			// Cross-sign their device with the self-signing key
+			if vh.mach.CrossSigningKeys != nil {
```

Comment:
This probably needs a check to make sure the user being verified is the same as the current user 🤔

Reply:
ok, I can look into that. Should we also add a `SignUser()` call in case it's not the same, like currently done in `ConfirmQRCodeScanned()`?

Reply:
Yeah, copying from the QR code flow is probably best if there's already an equivalent section there

Reply:
Picked it over, self-verifying still works. Not sure if I have a good way available to test the other path right now.